### DISCUSSION TITLE
To connect to multiple tables in a database

### DISCUSSION THREAD

**Discussion by sunyan-SG** [sinaptik-ai/pandas-ai#1149] [Ideas]
I checked the PandasAI documentation, it seems that it doesn't support on accessing multiple tables when connecting to a database. 
Is my understanding correct? if Yes, can pls add this new feature into PandasAI. If not, can pls show me how to specify multiple tables in database connector?  thank you.

**Comment by phoenixor** [comment]
Have you solved this problem? I also need to connect multiple MySQL data tables for joint operation, but the connector of the official example needs to be passed into the table, as if a table needs a connector.
![image](https://github.com/user-attachments/assets/792e3a0c-5aff-4901-9714-f03cc225e813)

**Comment by naredlarohithreddy** [comment]
@phoenixor  @sunyan-SG  guys , are you able to solve the problem, if not how did you proceed ? any alternatives

**Comment by gdcsinaptik** [comment]
@sunyan-SG, @phoenixor, and @naredlarohithreddy in the 3.0 there will be more flexibility on data operations via a semantic data layer. The feature you are suggesting here is about having a dataframe getting data from multiple sqp tables?

**Reply by naredlarohithreddy** [reply]
yeah, need to have multiple tables in sql database , so that based on the input, the llm must be able to perform joins on the tables and give the results.

**Comment by zev-yang** [comment]
May i know if it support the tables connection?  i tried it in the platform and gave clear instruction, like "connect dataset 'testing' and dataset 'testing2' with the key 'area', and sum the sales for 'person Zhang1"... but it doesn't work.

thanks.

**Comment by itd2007** [comment]
see this: https://github.com/sinaptik-ai/pandas-ai/blob/release/v2.4/examples/table_relations.py
